benchmark/domain/synthetic_3.py

```diff
@@ -0,0 +1,93 @@
+"""Synthetic dataset. Custom parabolic test with irrelevant parameters."""
+
+from uuid import UUID
+
+from numpy import sin, sqrt
+from pandas import DataFrame
+
+from baybe.campaign import Campaign
+from baybe.objective import SingleTargetObjective
+from baybe.parameters import NumericalContinuousParameter, NumericalDiscreteParameter
+from baybe.recommenders import RandomRecommender
+from baybe.searchspace import SearchSpace
+from baybe.simulation import simulate_scenarios
+from baybe.targets import NumericalTarget, TargetMode
+from benchmark.definition import Benchmark
+
+
+def lookup_synthetic_3(x: float, y: float, z: int) -> float:
+    """Synthetic dataset.
+
+        Number of Samples            inf
+        Dimensionality                 3
+        Features:
+            x   continuous [-2*pi, 2*pi]
+            y   continuous [-2*pi, 2*pi]
+            z   discrete {1,2,3,4}
+        Targets:
+            output   continuous
+    Best Value 4.09685
+    """
+    if z == 1:
+        return sin(x) * (1 + sin(y))
+    if z == 2:
+        return x * sin(0.9 * x) + sin(x) * sin(y)
+    if z == 3:
+        return sqrt(x + 8) * sin(x) + sin(x) * sin(y)
+    if z == 4:
+        return x * sin(1.666 * sqrt(x + 8)) + sin(x) * sin(y)
```

Comment:
remind me, what are these magic number like `1.666` again?

Reply:
I haven't created the function myself and thought that there is no further semantic beside scaling behind them. But your are right, should be defined if possible. Maybe @AVHopp can help here?

Reply:
I don't remember anymore. I unfortunately do not remember where this function was taken from, but I think there was not actual meaning behind the numbers. Iirc, this was just our first "Let's do something hybrid" test function. Happy to replace with something more reasonable.